foc_math.c

```diff
@@ -613,6 +613,9 @@ void foc_precalc_values(motor_all_state_t *motor) {
 	const mc_configuration *conf_now = motor->m_conf;
 	motor->p_lq = conf_now->foc_motor_l + conf_now->foc_motor_ld_lq_diff * 0.5;
 	motor->p_ld = conf_now->foc_motor_l - conf_now->foc_motor_ld_lq_diff * 0.5;
-	motor->p_inv_ld_lq = (1.0 / motor->p_lq - 1.0 / motor->p_ld);
-	motor->p_v2_v3_inv_avg_half = (0.5 / motor->p_lq + 0.5 / motor->p_ld) * 0.9; // With the 0.9 we undo the adjustment from the detection
+	motor->p_inv_inv_ld_lq = 1.0 / (1.0 / motor->p_lq - 1.0 / motor->p_ld);
+
+	motor->p_inv_lo_current_max = 1.0 / conf_now->lo_current_max;
```

Comment:
The lo-limits are special as they are updated in mc_interface when the different limits are hit, so they cannot be precalculated here. There should probably be some comment explaining that in the struct. This works because mcpwm_foc does not have it's own instance of the configuration - it is mc_interface that owns it and mcpwm_foc only has a pointer to it. The mcpwm_foc_set_configuration-function is only used to tell mcpwm_foc that the configuration has changed so that some checks and updates can be done on it if needed. This function is not called when the lo_limits are updated.

It might be worth rethinking this part at some point if there is a simpler/better solution.

Reply:
It might be worth moving things which are not explicitly user configuration out of the `conf` object and into a separate struct. The memory could be allocated in a similar way, so that there is only one instance shared globally.

I see how this variable is set in `update_override_limits()`. I think it would make sense to have these variables in a globally accessible struct which is written through setters. Modules could subscribe to update notifications, so they would recalculate only when the value has changed.